packages/cache/src/internal/tar.ts

```diff
@@ -117,31 +132,43 @@ export async function createTar(
   // --long=#: Enables long distance matching with # bits. Maximum is 30 (1GB) on 32-bit OS and 31 (2GB) on 64-bit.
   // Using 30 here because we also support 32-bit self-hosted runners.
   // Long range mode is added to zstd in v1.3.2 release, so we will not use --long in older version of zstd.
-  function getCompressionProgram(): string[] {
+  async function getCompressionProgram(): Promise<string[]> {
+    const tarPath = await getTarPath([])
+    const BSD_TAR_ZSTD = IS_WINDOWS && tarPath === SystemTarPathOnWindows
```

Comment:
Naming it `BSD_TAR_WINDOWS` may be better as it coveys that this applies only for windows and uses bsdtar. zstd is already evident from switch case below.

Reply:
Makes sense